Src/Support/Google.Apis.Auth/OAuth2/AwsExternalAccountCredential.AwsSignedSubjectToken.cs

```diff
@@ -0,0 +1,279 @@
+
+                string GetHashedCanonicalRequest()
+                {
+                    // Step 1
+                    // https://docs.aws.amazon.com/general/latest/gr/sigv4-create-canonical-request.html
+
+                    // First, the HTTP Method.
+                    // See point 1 in https://docs.aws.amazon.com/general/latest/gr/sigv4-create-canonical-request.html
+                    StringBuilder canonicalRequest = new StringBuilder(StVerificationHttpMethod).Append(newLine);
+
+                    // Now the URI's absolute path.
+                    // See point 2 in https://docs.aws.amazon.com/general/latest/gr/sigv4-create-canonical-request.html
+                    canonicalRequest.Append(verificationRequestUrl.GetPathForAwsCanonicalRequest()).Append(newLine);
+
+                    // Now the query string.
+                    // See point 3 in https://docs.aws.amazon.com/general/latest/gr/sigv4-create-canonical-request.html
+                    canonicalRequest.Append(verificationRequestUrl.GetQueryStringForAwsCanonicalRequest()).Append(newLine);
+
+                    // Now the canonical headers.
+                    // We just built them oursevels, so we know header names are lower case and that values
+                    // that could have leading or trailing spaces have been trimmed.
+                    // See point 4 in https://docs.aws.amazon.com/general/latest/gr/sigv4-create-canonical-request.html
+                    foreach (var headerPair in sortedHeaders)
+                    {
+                        canonicalRequest.Append(headerPair.Key);
+                        canonicalRequest.Append(":");
+                        canonicalRequest.Append(headerPair.Value);
+                        canonicalRequest.Append(newLine);
+                    }
+
+                    // Now the list of headers that will be signed.
+                    // See point 5 in https://docs.aws.amazon.com/general/latest/gr/sigv4-create-canonical-request.html
+                    canonicalRequest.Append(signedHeaders).Append(newLine);
+
+                    // Now the hashed request payload which is empty in our case.
+                    // See point 6 in https://docs.aws.amazon.com/general/latest/gr/sigv4-create-canonical-request.html
+                    canonicalRequest.Append(GetHexEncodedSha256Hash(Encoding.UTF8.GetBytes("")));
+
+                    // Point 7 in https://docs.aws.amazon.com/general/latest/gr/sigv4-create-canonical-request.html
+                    // is just to combine all of the previous elements into a single string, and we've been doing
+                    // that already in canonicalRequest.
+
+                    // Now hash and return the canonical request.
+                    // See point 8 in https://docs.aws.amazon.com/general/latest/gr/sigv4-create-canonical-request.html
+                    return GetHexEncodedSha256Hash(Encoding.UTF8.GetBytes(canonicalRequest.ToString()));
```

Comment:
Given that both calls to this function encode a string, perhaps change the method to accept a string and do the encoding within it?